components/callbacks.py

```diff
@@ -225,196 +217,114 @@ def off_on_topic(update: Update, context: CallbackContext) -> None:
-    result = extract_status_change(chat_member)
-    if result is None:
-        return
+async def delete_new_chat_members_message(update: Update, _: ContextTypes.DEFAULT_TYPE) -> None:
+    await cast(Message, update.effective_message).delete()
 
-    # Only greet members how newly joined the group
-    was_member, is_member = result
-    if not (was_member is False and is_member is True):
-        return
 
-    # Get groups name
-    group_user_name = cast(str, cast(Chat, update.effective_chat).username)
+async def leave_chat(update: Update, context: ContextTypes.DEFAULT_TYPE) -> None:
+    context.application.create_task(cast(Chat, update.effective_chat).leave(), update=update)
+    raise ApplicationHandlerStop
 
-    # Just a precaution in case the bot was added to a different group
-    if group_user_name not in [ONTOPIC_USERNAME, OFFTOPIC_USERNAME]:
-        return
 
-    # Get saved users
-    chat_data = cast(dict, context.chat_data)
-    user_lists = chat_data.setdefault("new_chat_members", {})
-    users = user_lists.setdefault(group_user_name, [])
+async def raise_app_handler_stop(update: Update, _: ContextTypes.DEFAULT_TYPE) -> None:
+    raise ApplicationHandlerStop
 
-    # save new user
-    users.append(chat_member.new_chat_member.user.mention_html())
 
-    # check rate limit
-    last_message_date = chat_data.setdefault(
-        "new_chat_members_timeout",
-        dtm.datetime.now() - dtm.timedelta(minutes=NEW_CHAT_MEMBERS_LIMIT_SPACING + 1),
-    )
-    next_possible_greeting_time = last_message_date + dtm.timedelta(
-        minutes=NEW_CHAT_MEMBERS_LIMIT_SPACING
-    )
-    if dtm.datetime.now() < next_possible_greeting_time:
-        # We schedule a job to the next possible greeting time so that people are greeted
-        # and presented with the rules as early as possible while not exceeding the rate limit
-        logging.debug("Scheduling job to greet new members after greetings-cool down.")
-        job_queue = cast(JobQueue, context.job_queue)
-        jobs = job_queue.get_jobs_by_name("greetings_job")
-        if not jobs:
-            job_queue.run_once(
-                callback=lambda _: do_greeting(
-                    bot=context.bot,
-                    chat_data=chat_data,
-                    group_user_name=group_user_name,
-                    users=users,
-                ),
-                when=(next_possible_greeting_time - dtm.datetime.now()).seconds,
-                name="greetings_job",
-            )
-    else:
-        do_greeting(
-            bot=context.bot, chat_data=chat_data, group_user_name=group_user_name, users=users
-        )
-
-
-def list_available_hints(update: Update, _: CallbackContext) -> None:
+async def list_available_hints(update: Update, _: ContextTypes.DEFAULT_TYPE) -> None:
     private = False
     if cast(Chat, update.effective_chat).type != Chat.PRIVATE:
         text = "Please use this command in private chat with me."
```

Comment:
```suggestion
    group = False
    if cast(Chat, update.effective_chat).type != Chat.PRIVATE:
        text = "Please use this command in private chat with me."
```
then inside the if condition change `group = True`. Didn't make sense to me to set `private=True` when the chat is not private

Reply:
actuall that function is not used anywhere … removing